lib/reporter.coffee

```diff
@@ -44,12 +29,43 @@ failOnWarningReporter = ->
             createPluginError "CoffeeLint failed for #{file.relative}"
         cb()
 
+loadReporter = (reporter) ->
+    return reporter if typeof reporter is 'function'
+    if typeof reporter is 'object' and typeof reporter.reporter is 'function'
+        return reporter.reporter
+
+    if typeof reporter is 'string'
+        # Try to load CoffeeLint's build-in reporters
+        try
+            return loadReporter require('coffeelint/src/reporters/' + reporter)
```

Comment:
Requiring `.coffee` files would make the `coffee-script` npm module a production dependency, which is really bad style for a npm module.